bxt-strafe/src/lib.rs

```diff
@@ -81,6 +83,11 @@ pub struct Parameters {
     pub step_size: f32,
     pub bounce: f32,
     pub bhop_cap: bool,
+    pub bhop_cap_multiplier: f32,
+    pub bhop_cap_maxspeed_scale: f32,
+    pub use_slow: bool,
+    pub has_stamina: bool,
```

Comment:
I wonder if maybe we can just get rid of this variable, and set stamina_time = 0 for games without stamina?

Reply:
Not sure what you mean by this. There is currently this line responsible for updating stamina.

https://github.com/YaLTeR/bxt-rs/blob/a8e93b22cefe5cda17da3f221cefb1a8295d92a1/src/modules/tas_optimizer/mod.rs#L864

However, it just updates. The variable `has_stamina` makes it meaningful. Otherwise, I guess we have to use the `com_gamedir` to says that stamina only applies to that game? I guess if you want that then sure.

Reply:
Yeah, I was thinking put the game dir check here, and set to 0 if it's not cstrike.

Reply:
Hmmm, from my testing, removing this would be more trouble than it's worth. This would also require checking game in the `bxt-strafe/../step.rs` as well. I attempted to just check whether the stamina is there (stamina is 0) and do actions accordingly but it is possible for CS1.6 to have stamina of 0 too.

Reply:
This just makes the stamina prediction just weirdly borked.

Reply:
I guess I have to do actual checking of the game then?

Reply:
I thought all stamina operations just do nothing when stamina is 0, requiring no extra checks? Or am I mistaken

Reply:
Welp, that would complicate things too as it needs thread marker. Would you reconsider this?

Reply:
Ah, right, the stamina set on jump. Forgot about that. Yeah let's just leave this as is then

Reply:
Chat goes so fast that Github doesn't even refresh :+1: maybe it is my adblocker.